src/structarray.jl

```diff
@@ -365,6 +365,12 @@ function Base.push!(s::StructVector, vals)
     return s
 end
 
+function Base.pop!(s::StructVector)
+    out = s[end]
+    foreachfield(pop!, s)
+    return out
+end
```

Comment:
```suggestion
function Base.pop!(s::StructVector{T}) where T
    t = map(pop!, components(s))
    return createinstance(T, t...)
end
```

This may be a bit cleaner, as you only traverse the arrays once.